conda/core/path_actions.py

```diff
@@ -1077,8 +1077,15 @@ def execute(self):
                 #   record that url as the remote source url in urls.txt
                 # we do the search part of this operation before the create_link so that we
                 #   don't md5sum-match the file created by 'create_link'
+                # there is no point in looking for the tarball in the cache that we are writing
+                #   this file into because we have already removed the previous file if there was
+                #   any. This also makes sure that we ignore the md5sum of a possible extracted
+                #   directory that might exist in this cache because we are going to overwrite it
+                #   anyway when we extract the tarball.
                 source_md5sum = compute_md5sum(source_path)
-                pc_entry = PackageCache.tarball_file_in_cache(source_path, source_md5sum)
+                exclude_caches = [self.target_pkgs_dir]
```

Comment:
Also might as well make this a tuple.

    exclude_caches = self.target_pkgs_dir,